docs_src/microservices/application/AdvancedTopics.md

```diff
@@ -6,261 +6,234 @@ The following items discuss topics that are a bit beyond the basic use cases of
+To add your own route, use the `AddRoute()` API provided on the `ApplicationService` interface. 
 
-For usages where the incoming data is not `events`, the `TargetType` of the excepted incoming data can be set when the SDK instance is created. 
+!!! example  "Example - Add Custom REST route"
 
-Example:
+    ``` go      
+    myhandler := func(writer http.ResponseWriter, req *http.Request) {    
+      service := req.Context().Value(interfaces.AppServiceContextKey).(interfaces.ApplicationService)    
+      service.LoggingClient().Info("TEST")     
+      writer.Header().Set("Content-Type", "text/plain")   
+      writer.Write([]byte("hello"))   
+      writer.WriteHeader(200)    
+    }    
+    
+    service := pkg.NewAppService(serviceKey)    
+    service.AddRoute("/myroute", myHandler, "GET")    
+    ```    
 
-``` go
-type Person struct {
-    FirstName string `json:"first_name"`
-    LastName  string `json:"last_name"`
-}
+Under the hood, this simply adds the provided route, handler, and method to the gorilla `mux.Router` used in the SDK. For more information on `gorilla mux` you can check out the github repo [here](https://github.com/gorilla/mux). 
+You can access the `interfaces.ApplicationService` API for resources such as the logging client by pull it from the context as shown above -- this is useful for when your routes might not be defined in your main.go, where you have access to the ``interfaces.ApplicationService`` instance.
 
-edgexSdk := &appsdk.AppFunctionsSDK {
-	ServiceKey: serviceKey, 
-	TargetType: &Person{},
-}
-```
+### Target Type
 
-`TargetType` must be set to a pointer to an instance of your target type such as `&Person{}` . The first function in your function pipeline will be passed an instance of your target type, not a pointer to it. In the example above, the first function in the pipeline would start something like:
+The target type is the object type of the incoming data that is sent to the first function in the function pipeline. By default this is an EdgeX `dtos.Event` since typical usage is receiving `Events` from the EdgeX MessageBus. 
 
-``` go
-func MyPersonFunction(edgexcontext *appcontext.Context, params ...interface{}) (bool, interface{}) {
+There are scenarios where the incoming data is not an EdgeX `Event`. One example scenario is two application services are chained via the EdgeX MessageBus. The output of the first service, inference data from analyzing the original `Event`data, is published back to the EdgeX MessageBus. The second service needs to be able to let the SDK know the target type of the input data it is expecting.
```

Comment:
The second sentence a bit segmented and hard to read at the first glance, suggest re-writting a bit.
```suggestion
There are scenarios where the incoming data is not an EdgeX `Event`. One example scenario is two application services are chained via the EdgeX MessageBus. The output of the first service is inference data from analyzing the original `Event`data, and published back to the EdgeX MessageBus. The second service needs to be able to let the SDK know the target type of the input data it is expecting.
```